catavatar/catavatar.php

```diff
@@ -92,36 +95,29 @@ function catavatar_addon_settings_post(App $a, &$s)
 	if (!empty($_POST['catavatar-usecat'])) {
 		$url = $a->get_baseurl() . '/catavatar/' . local_user();
 
-		// set the catavatar url as avatar url in contact and default profile
-		// and set profile to 0 to current photo
-		// I'm not sure it's the correct way to do this...
-		$r = dba::update('contact',
-			['photo' => $url . '/4', 'thumb' => $url . '/5', 'micro' => $url . '/6', 'avatar-date' => DateTimeFormat::utcNow()],
-			['uid' => local_user(), 'self' => 1]
-		);
-		if ($r===false) {
-			notice(L10n::t('There was an error, the cat ran away.'));
+		$self = dba::selectFirst('contact', ['id'], ['uid' => local_user(), 'self' => true]);
+		if (!DBM::is_result($self)) {
+			notice(L10n::t("The cat hadn't found itself."));
```

Comment:
Can you please re-generate the messages.po file for the addon as you added a new string to the translation pipe.

Reply:
Only if you tell me, how I do this :-)

Reply:
Go into the directory of the addon you want to extract the strings from and then call the `run_xgettext` tool from the Friendica tools with the parameter `-s`. So in this case

    cd /path/to/friendica-addons/catavatar
    /path/to/friendica/bin/run_xgettext.sh -s

should let the magic happen.